cmd/livepeer/livepeer.go

```diff
@@ -200,13 +214,33 @@ func main() {
 		n.EthPassword = *ethPassword
 
 		if *transcoder {
-			logsSub, err := setupTranscoder(n, acct)
+			registered, err := n.Eth.IsRegisteredTranscoder()
+			if err != nil {
+				glog.Errorf("Error checking for registered transcoder: %v", err)
+			}
 
+			logsCh := make(chan ethtypes.Log)
+			logsSub, err := n.Eth.SubscribeToJobEvent(context.Background(), logsCh)
 			if err != nil {
 				glog.Errorf("Error subscribing to job event: %v", err)
 			}
+
+			defer close(logsCh)
 			defer logsSub.Unsubscribe()
-			// defer close(logsChan)
+
+			if !registered {
+				fmt.Println("You are not registered as a transcoder. To register and become a candidate transcoder please bond some tokens: ")
+
+				var bondAmount int
+				_, err := fmt.Scanf("%d", &bondAmount)
+				if err != nil {
+					glog.Errorf("Error reading bond amount from input: %v", err)
+				}
+
+				registerAndSetupTranscoder(n, logsCh, big.NewInt(int64(bondAmount)), uint8(*blockRewardCut), uint8(*feeShare), big.NewInt(int64(*pricePerSegment)))
+			} else {
+				setupTranscoder(n, logsCh)
+			}
```

Comment:
I think we should exit if `registerAndSetupTranscoder` or `setupTranscoder` fail